### PR TITLE
NIRISS APT interface

### PR DESCRIPTION
**Pull request by Johannes-Sahlmann**
This is the first functional version that supports processing a basic NIRISS APT file (#73).
This needs careful review because many changes were made and I did not have an easy way to verify that they don't break the use for NIRCam.

Because NIRISS and NIRCam reference files are located in different locations and have different names, changes to the SimInput __init__ method were made, which imply that mirage can only work on the observations of one instrument at the time (instrument can be 'WFSC', 'NIRCam', ' NIRISS').

If needed I can make the example driver script and the APT file available (eventually we want to have this as a test case).

### CODE REVIEW

**Comment by bhilbert4** [comment]
I think it's good to get NIRISS working here, and then later we can think about APT files that include multiple instruments. For now, if there are multiple instruments, then all inputs should be generated by running the APT file through the code once for each instrument, right? 

I can try running a NIRCam APT file using this branch and make sure the outputs look ok. I don't think I've ever generated an APT file with prime and parallel instruments, but assuming I can do that, I can test that case as well.

**Comment by Johannes-Sahlmann** [comment]
The answer to your question is yes.

It would be great if you test this branch in the way you suggest, which bridges the gap to the tests that Lauren is setting up (#64)

**Comment by Johannes-Sahlmann** [comment]
Yes, please send me that NIRCam APT file and I will fix that.

**Comment by Johannes-Sahlmann** [comment]
The NIRCam example files that Bryan gave me now pass successfully. Note that the fix does not yet include any NIRCam template other than the basic imaging one. I deferred additional changes until we decide if/how to restructure read_apt_xml. 

@laurenmarietta if you want to keep the current structure of read_apt_xml for the time being, I am happy to implement the fix in all the read_template functions. However, this will be a blind change, as I don't have a way to test.

**Comment by Johannes-Sahlmann** [comment]
Even if currently only very basic modes of NIRISS are supported, I think this is ready to merge (from my side). 
More advanced and complex use cases will be much easier to develop in new branches ones we have the test infrastructure set up.

File: nircam_simulator/scripts/catalog_seed_image.py [hunk 1]
```diff
@@ -1637,6 +1637,10 @@ def makePointSourceImage(self, pointSources):
             if bstr == "0.0":
                 bstr = "0.00"
 
+            if self.params['simSignals']['psfbasename'].lower() == 'niriss':
```
Comment (bhilbert4) [inline nircam_simulator/scripts/catalog_seed_image.py]: I think you may need to rebase this from `master`. I merged some changes on Thursday that solved the PSF filename issue, such that two digits are always kept from the sub-pixel position.

Comment (bhilbert4) [reply nircam_simulator/scripts/catalog_seed_image.py]: Rebasing may also fix the conflicts below.

**Comment by Johannes-Sahlmann** [comment]
Rebasing seemed overkill given the minor merge conflict. I resolved it by using the master branch code. I believe your changes to the master also solved the problem with identifying the WebbPSF library file for NIRISS. I will test this after the merge.

**Pull request merged**